File: Kotlin/app/src/main/java/dk/coded/emia/View/pages/MyPostsFragment.java
package dk.coded.emia.View.pages;

import android.app.Activity;

import dk.coded.emia.View.fragment.PostListFragment;
import dk.coded.emia.model.Data.Post;
import dk.coded.emia.model.Data.User;
import dk.coded.emia.model.adapter.PostListValidator;
import dk.coded.emia.utils.BasicCallBack;
import dk.coded.emia.model.interactor.DatabaseFactory;
import dk.coded.emia.model.interactor.DatabaseInteractor;

import static dk.coded.emia.utils.Constants.SUCCESS;

public class MyPostsFragment extends PostListFragment {

    private DatabaseInteractor mInteractor;

    public MyPostsFragment() {
        mInteractor = DatabaseFactory.INSTANCE.getDatabaseInteractor();
    }

    @Override
    public void needShow(Post post, User user, BasicCallBack callback) {
        if (!getMValidator().isValid(post, user)) {
            callback.callBack(Companion.getSUCCESS(), false);
        } else {
            String currentUserId = mInteractor.getCurrentUserId();
            Boolean isItMyPost = post.getUid().equals(currentUserId);
            callback.callBack(Companion.getSUCCESS(), isItMyPost);
        }
    }
}

File: Kotlin/app/src/main/java/dk/coded/emia/View/pages/MyPostsFragment.kt
package dk.coded.emia.View.pages

import android.app.Activity

import dk.coded.emia.View.fragment.PostListFragment
import dk.coded.emia.model.Data.Post
import dk.coded.emia.model.Data.User
import dk.coded.emia.model.adapter.PostListValidator
import dk.coded.emia.utils.BasicCallBack
import dk.coded.emia.model.interactor.DatabaseFactory
import dk.coded.emia.model.interactor.DatabaseInteractor
import dk.coded.emia.utils.Constants

class MyPostsFragment : PostListFragment() {

    private val mInteractor: DatabaseInteractor

    init {
        mInteractor = DatabaseFactory.databaseInteractor
    }

    override fun needShow(post: Post, user: User, callback: BasicCallBack) {
        if ((mValidator!!.isValid(post, user))!! == false) {
            callback(Constants.SUCCESS, false)
        } else {
            val currentUserId = mInteractor.currentUserId
            val isItMyPost = post.uid == currentUserId
            callback(Constants.SUCCESS, isItMyPost)
        }
    }
}
